### ISSUE TITLE
Find-VulnerableDnsRecords doesn't work if not launched on a DNS server

### ISSUE THREAD

**Issue by cbo2077** [akamai/Invoke-DHCPCheckup#3]
as from the tile, the function Find-VulnerableDnsRecords assume that the script is launched from a DNS server

**Comment by amccann-mbari** [comment]
I ran into the same issue. It seems like this script is assuming that it is running on a domain controller that is also acting as a DHCP server.

It seems like the only part of this script that needs to directly target a DNS server is in the `Find-VulnerableDnsRecords` function. Adding a command to discover domain controllers based on the domain name provided when calling the script then using that discovered DC to query DNS records seems to work around the issue.

```diff
+$DnsServer = Get-ADDomainController -DomainName $domainName -Discover | Select-Object -ExpandProperty HostName
+$DnsRecords = Get-DnsServerResourceRecord -ZoneName $domainName -ComputerName $DnsServer
-$DnsRecords = Get-DnsServerResourceRecord -ZoneName $domainName
```

There is probably a better way to go about this that would make the entire script run on any system as long as the appropriate PowerShell modules are installed, but this is the simplest way to just to get it to work. You still need to run this script directly on a DHCP server but at least it would work if your DNS is installed somewhere else.

**Comment by gc-oridavid** [comment]
Solved this problem with #7, enabling to specify the name of the DNS server manually.
@amccann-mbari - thanks for the suggestion, I would be glad to use some sort of automation instead of an argument, but I'm afraid this approach might not always work (if there are multiple DCs for example, where only one is a DNS server).
